src/assets/wise5/components/table/tableService.ts

```diff
@@ -147,28 +147,39 @@ export class TableService extends ComponentService {
     if (componentState != null) {
       const studentData = componentState.studentData;
       if (studentData != null) {
-        const studentTableData = studentData.tableData;
-        const componentContentTableData = componentContent.tableData;
-        if (studentTableData != null) {
-          const studentRows = studentTableData;
-          for (let r = 0; r < studentRows.length; r++) {
-            const studentRow = studentRows[r];
-            if (studentRow != null) {
-              for (let c = 0; c < studentRow.length; c++) {
-                const studentCell = this.getTableDataCellValue(r, c, studentTableData);
-                const componentContentCell = this.getTableDataCellValue(
-                  r,
-                  c,
-                  componentContentTableData
-                );
-                if (studentCell !== componentContentCell) {
-                  /*
-                   * the cell values are not the same which means
-                   * the student has changed the table
-                   */
-                  return true;
-                }
-              }
+        return (
+          this.studentDataHasSelectedRows(studentData) ||
+          this.studentDataHasTableData(studentData, componentContent)
+        );
+      }
+    }
+  }
+
+  private studentDataHasSelectedRows(studentData: any): boolean {
+    return studentData.selectedRowIndices != null;
+  }
+
+  private studentDataHasTableData(studentData: any, componentContent: any): boolean {
+    const studentTableData = studentData.tableData;
+    const componentContentTableData = componentContent.tableData;
+    if (studentTableData != null) {
```

Comment:
Looks like ```componentContentTableData``` is only used if studentTableData is not null, so move it inside the if block
```suggestion
    if (studentTableData != null) {
      const componentContentTableData = componentContent.tableData;    
```